### DISCUSSION TITLE
Example: NimBLE_server_get_client_name

### DISCUSSION THREAD

**Discussion by mignon-ne** [h2zero/NimBLE-Arduino#741] [Q&A] [answered]
Hello,
I have NimBLE as Server on my ESP32S3 - when mobile phone connected to my Server I would like to have their name.
Came across to this example but cannot comply as error below:

> error: 'void ServerCallbacks::onConnect(NimBLEServer*, NimBLEConnInfo&, std::string&)' marked 'override', but does not override
>    23 |         void onConnect(NimBLEServer* pServer, NimBLEConnInfo& connInfo, std::string& name) override {
>       |                 ^~~~~~~~~
> 
> and
> 
> error: 'class NimBLEServer' has no member named 'getPeerNameOnConnect'
>    66 |         pServer->getPeerNameOnConnect(true); // Setting this will enable the onConnect callback that provides the name.
> 

Check with the src/NimBLEServer.cpp, seems difference from my current installed Library 1.4.2.
Any advice how to get the Client name or need a library update but latest is 1.4.2.

Thank you in advance.

**Comment by h2zero** [comment]
This only works with the master branch currently. It will be available in the next release.

**Reply by mignon-ne** [reply]
Thank you, in that case may you advice how to get the client name with current release.

**Reply by h2zero** [reply, marked as answer]
There is no easy way to do this in the current release, you would need to use lower level calls. Have a look at the related functions in the master branch if you wish to attempt this.

**Reply by mignon-ne** [reply]
Thank you.
